src/scopes.rs

```diff
@@ -41,1046 +73,210 @@ pub enum ScopeKind {
-    assert!(module_scope.get_binding("export2").is_none());
-    assert!(module_scope.get_binding("alias1").is_some());
-    assert!(module_scope.get_binding("export3").is_some());
-    assert!(module_scope.get_binding("export4").is_some());
-    assert!(module_scope.get_binding("export5").is_some());
-    assert!(module_scope.get_binding("export6").is_none());
-    assert!(module_scope.get_binding("alias2").is_some());
-    assert!(module_scope.get_binding("defaultExport2").is_some());
-    assert!(module_scope.get_binding("export7").is_some());
-    assert!(module_scope.get_binding("defaultExport3").is_some());
-    assert!(module_scope.get_binding("namespaced2").is_some());
-  }
-
-  #[test]
-  fn destructuring_assignment() {
-    let source_code = r#"
-const {a} = {a : "a"};
-const {a: {b}} = {a : {b: "b"}};
-const {a: {b: {c}}} = {a : {b: {c : "c"}}};
-const [d] = ["d"];
-const [e, [f,[g]]] = ["e",["f",["g"]]];
-const {a: [h]} = {a: ["h"]};
-const [i, {j}] = ["i",{j: "j"}];
-const {a: {b : [k,{l}]}} = {a: {b : ["k",{l : "l"}]}};
-const {m = "M"} = {};
-const [n = "N"] = [];
-const {...o} = {o : "O"};
-const [...p] = ["p"];
-function getPerson({username="disizali",info: [name, family]}) {
-  try {
-    throw 'TryAgain';
-  } catch(e) {}
-}
-try{} catch({message}){};
-"#;
-    let root_scope = test_scopes(source_code);
-    let module_scope = root_scope.get_child_scope(0).unwrap();
-    assert!(module_scope.get_binding("a").is_some());
-    assert!(module_scope.get_binding("b").is_some());
-    assert!(module_scope.get_binding("c").is_some());
-    assert!(module_scope.get_binding("d").is_some());
-    assert!(module_scope.get_binding("e").is_some());
-    assert!(module_scope.get_binding("f").is_some());
-    assert!(module_scope.get_binding("g").is_some());
-    assert!(module_scope.get_binding("h").is_some());
-    assert!(module_scope.get_binding("i").is_some());
-    assert!(module_scope.get_binding("j").is_some());
-    assert!(module_scope.get_binding("k").is_some());
-    assert!(module_scope.get_binding("l").is_some());
-    assert!(module_scope.get_binding("m").is_some());
-    assert!(module_scope.get_binding("n").is_some());
-    assert!(module_scope.get_binding("o").is_some());
-    assert!(module_scope.get_binding("p").is_some());
-
-    let function_scope = module_scope.get_child_scope(0).unwrap();
-    assert!(function_scope.get_binding("username").is_some());
-    assert!(function_scope.get_binding("name").is_some());
-    assert!(function_scope.get_binding("family").is_some());
-
-    let function_catch_scope = function_scope.get_child_scope(-1).unwrap();
-    assert!(function_catch_scope.get_binding("e").is_some());
+  fn visit_switch_stmt(&mut self, n: &SwitchStmt, _: &dyn Node) {
+    n.discriminant.visit_with(n, self);
 
-    let catch_scope = module_scope.get_child_scope(-1).unwrap();
-    assert!(catch_scope.get_binding("message").is_some());
+    self.visit_with_path(ScopeKind::Switch, &n.cases);
   }
 }
```

Comment:
Sad to see tests go, but we can add them later

Reply:
Oh.. I missed the assertion. I'll see if it's possible to restore.